### PR TITLE
Update 'New Addon' function to work with 4.2

### PR DESCRIPTION
**Pull request by strike-digital**
Changes:
- Added a new dialogue that asks whether the addon should support legacy Blender versions (<4.2).
- If not then the `bl_info` dict is removed from the `__init__.py` file.
- A manifest is now always copied into the new directory.
- Updated `auto_load.py` to work in 4.2
- Fixed bug in `auto_load` that would prevent it from recognising classes that didn't directly inherit from a Blender class, even if they did so indirectly (Could be removed, or split into a different pull request).

This assumes that all addons created from now will be extensions, and so making the addon backwards compatible is a choice.
Although maybe it should just be assumed that this is wanted?

### CODE REVIEW

File: src/new_addon.ts [hunk 1]
```diff
@@ -6,22 +6,23 @@ import { letUserPickItem } from './select_utils';
...
     addFolderToWorkspace, multiReplaceText, pathExists,
-    isValidPythonModuleName, renamePath
+    isValidPythonModuleName, renamePath, toTitleCase
 } from './utils';
 
-type AddonBuilder = (path: string, addonName: string, authorName: string) => Promise<string>;
+type AddonBuilder = (path: string, addonName: string, authorName: string, supportLegacy: boolean) => Promise<string>;
 
 const addonTemplateDir = path.join(templateFilesDir, 'addons');
+const manifestFile = path.join(addonTemplateDir, '..', 'blender_manifest.toml')
```
Comment (JacquesLucke) [inline src/new_addon.ts:15]: Could also join `templateFilesDir` with `blender_manifest.toml`  instead of having to use `..`?

File: src/new_addon.ts [hunk 2]
```diff
@@ -6,22 +6,23 @@ import { letUserPickItem } from './select_utils';
...
-    let [addonName, authorName] = await askUser_SettingsForNewAddon();
+    let [addonName, authorName, supportLegacy] = await askUser_SettingsForNewAddon();
     let folderPath = await getFolderForNewAddon();
     folderPath = await fixAddonFolderName(folderPath);
-    let mainPath = await builder(folderPath, addonName, authorName);
+    let mainPath = await builder(folderPath, addonName, authorName, supportLegacy);
 
     await vscode.window.showTextDocument(vscode.Uri.file(mainPath));
-    await vscode.commands.executeCommand('cursorBottom');
+    // await vscode.commands.executeCommand('cursorBottom');
```
Comment (JacquesLucke) [inline src/new_addon.ts:25]: Why this change? If it should be changed, then the line should be removed instead.

**Review by JacquesLucke — CHANGES_REQUESTED** [review]
Only got some minor comments. Generally looks good already, thanks!

File: src/new_addon.ts (hunk 2 above)
Comment (strike-digital) [reply src/new_addon.ts:25]: Since the manifest is now opened instead of the init file, moving the cursor to the bottom would hide the important info to change off the top of the screen. I just forgot to remove this line.

**Review by JacquesLucke — APPROVED** [review]
